tests/integration/conftest.py

```diff
@@ -202,17 +203,23 @@ async def app_fixture(
         model.add_relation(app_name, "nginx-ingress-integrator"),
     )
     await model.wait_for_idle(status="active")
+
+    # Enable plugins calling rake site_settings:import in one of the units.
     inline_yaml = "\n".join(f"{plugin}_enabled: true" for plugin in ENABLED_PLUGINS)
-    enable_plugins_command = (
-        "pebble exec --user=_daemon_ --context=discourse -w=/srv/discourse/app -ti -- /bin/bash -c "
-        f""""echo '{inline_yaml}' | """
-        '''/srv/discourse/app/bin/bundle exec rake site_settings:import -"'''
+    discourse_rake_command = "/srv/discourse/app/bin/bundle exec rake site_settings:import "
+    pebble_exec = (
+        "PEBBLE_SOCKET=/charm/containers/discourse/pebble.socket "
+        "pebble exec --user=_daemon_ --context=discourse -w=/srv/discourse/app"
     )
-
-    logger.info("Enabling plugins: %s", enable_plugins_command)
-    action = await unit.run(f"/bin/bash -c '{enable_plugins_command}'")
+    full_command = (
+        "/bin/bash -c "
+        f"'set -euo pipefail; echo \"{inline_yaml}\" | {pebble_exec} -- {discourse_rake_command}'"
+    )
+    logger.info("Enable plugins command: %s", full_command)
+    action = await unit.run(full_command)
```

Comment:
The command will run in the charm container. But it will exec in the discourse container (the pebble client in the charm container will talk to the pebble server in the discourse container thanks to the PEBBLE_SOCKET env var)